ansible-scylla-node/tasks/common.yml

```diff
@@ -299,6 +299,45 @@
       when: token_distributor is defined
   when: start_scylla_service is defined and start_scylla_service|bool
 
+- name: Check if authentication is enabled
+  lineinfile:
+    path: /etc/scylla/scylla.yaml
+    regexp: '^(authenticator: (.*PasswordAuthenticator.*|.*TransitionalAuthenticator.*))'
+    state: absent
+  check_mode: yes
+  register: _authentication_enabled
+  run_once: true
+
+- name: If authentication is enabled, adjust replication for system_auth keyspace
+  block:
+    - name: Get dc list
+      set_fact:
+        dcs_to_rf: "{{ dcs_to_rf | default([]) + [\"'\" + hostvars[item]['dc'] + \"':\" + replication_factor|string] }}"
+      loop: "{{ groups['scylla'] }}"
+      run_once: true
+
+    - name: Adjust replication for system_auth keyspace
+      shell: |
+        cqlsh {{ broadcast_address }} -u {{ cql_username }} -p {{ cql_password }} -e "ALTER KEYSPACE system_auth WITH replication = {'class': '{{ replication_strategy }}', {{ dcs_to_rf | unique | join(',') }}};"
+      run_once: true
+
+    - name: Repair system_auth
+      include_tasks: repair.yml
+      vars:
+        keyspace: 'system_auth'
+
+    - name: Cleanup system_auth
+      uri:
+        url: "http://{{ scylla_api_address }}:{{ scylla_api_port }}/storage_service/keyspace_cleanup/system_auth"
```

Comment:
Isn't this API blocking?

Reply:
Yes, but now I'm using the builtin ansible async task

Reply:
Fixed